app/Notifications/NewComment.php

```diff
@@ -47,6 +48,61 @@ public function via(object $notifiable): array
         return ['database'];
     }
 
+    /**
+     * Determine if the notification should be sent.
+     *
+     * @return bool
+     */
+    public function shouldSend(User $notifiable): bool
+    {
+        // Evaluate general settings
+        if ($this->comment->user_id === $notifiable->id || $notifiable->notification?->block_notifications == 1) {
+            return false;
+        }
+
+        // Initialize target variables
+        $targetGroup = '';
+        $targetSetting = '';
+
+        // Evaluate model based settings
+        switch (true) {
+            case $this->model instanceof Torrent:
+                $targetGroup = 'json_torrent_groups';
+                $targetSetting = 'show_torrent_comment';
+
+                break;
+            case $this->model instanceof TorrentRequest:
+                $targetGroup = 'json_request_groups';
+                $targetSetting = 'show_request_comment';
+
+                break;
+            case $this->model instanceof Ticket:
+                return ! ($this->model->staff_id === $this->comment->id && $this->model->staff_id !== null)
+                ;
+
+            case $this->model instanceof Playlist:
+            case $this->model instanceof Article:
+                return true;
+            case $this->model instanceof Collection:
+                break;
```

Comment:
It's not possible to notify on collection comments since they aren't owned by users. Probably best to remove this case, and, remove the Collection type from the constructor, and remove the message at the bottom too.